File: 0058-length-of-last-word/0058-length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    const a = s.trim();
    const words = a.split(/\s+/);
    const lastWord = words[words.length - 1];
    return lastWord.length;
};

File: 0058-length-of-last-word/0058-length-of-last-word.go
func lengthOfLastWord(s string) int {
	length := 0
	inWord := false

	for i := len(s) - 1; i >= 0; i-- {
		if s[i] != ' ' {
			length++
			inWord = true
		} else if inWord {
			break
		}
	}

	return length
}

